factory/config.go

```diff
@@ -79,22 +81,27 @@ func addClusterConfig(
 		c.ClusterType = terraform.GetVariableAsStringFromVarFile(g, varDir, "cluster_type")
 		c.ExternalDb = terraform.GetVariableAsStringFromVarFile(g, varDir, "external_db")
 		c.RenderedTemplate = terraform.Output(g, terraformOptions, "rendered_template")
-		shared.KubeConfigFile = "/tmp/" + terraform.Output(g, terraformOptions, "kubeconfig") + "_kubeconfig"
-	} else {
-		shared.KubeConfigFile = terraform.Output(g, terraformOptions, "kubeconfig")
-	}
+	} 
+	
+	shared.KubeConfigFile = terraform.Output(g, terraformOptions, "kubeconfig")
+	shared.AwsUser = terraform.GetVariableAsStringFromVarFile(g, varDir, "aws_user")
+	shared.AccessKey = terraform.GetVariableAsStringFromVarFile(g, varDir, "access_key")
+	c.ProductType = cfg.Product
+	
+	serverIPs := strings.Split(terraform.Output(g, terraformOptions, "master_ips"), ",")
+	c.ServerIPs = serverIPs
 
 	rawAgentIPs := terraform.Output(g, terraformOptions, "worker_ips")
 	if rawAgentIPs != "" {
 		agentIPs = strings.Split(rawAgentIPs, ",")
 	}
 	c.AgentIPs = agentIPs
 
-	ServerIPs := strings.Split(terraform.Output(g, terraformOptions, "master_ips"), ",")
-	c.ServerIPs = ServerIPs
-
-	shared.AwsUser = terraform.GetVariableAsStringFromVarFile(g, varDir, "aws_user")
-	shared.AccessKey = terraform.GetVariableAsStringFromVarFile(g, varDir, "access_key")
+	rawAgentIPs = terraform.Output(g, terraformOptions, "windows_worker_ips")
+	if rawAgentIPs != "" {
+		agentIPs = strings.Split(rawAgentIPs, ",")
+	}
+	c.WinAgentIPs = agentIPs
```

Comment:
this logic here will not work when u are not creating windowns ips.....

you should do something like
```
func addClusterConfig(
	cfg *config.ProductConfig,
	g GinkgoTInterface,
	varDir string,
	terraformOptions *terraform.Options,
) (*Cluster, error) {
	c := &Cluster{}
	var agentIPs []string
	var winAgentIPs []string

	if cfg.Product == "k3s" {
		c.ClusterType = terraform.GetVariableAsStringFromVarFile(g, varDir, "cluster_type")
		c.ExternalDb = terraform.GetVariableAsStringFromVarFile(g, varDir, "external_db")
		c.RenderedTemplate = terraform.Output(g, terraformOptions, "rendered_template")
	}

	shared.KubeConfigFile = terraform.Output(g, terraformOptions, "kubeconfig")
	shared.AwsUser = terraform.GetVariableAsStringFromVarFile(g, varDir, "aws_user")
	shared.AccessKey = terraform.GetVariableAsStringFromVarFile(g, varDir, "access_key")
	c.ProductType = cfg.Product

	serverIPs := strings.Split(terraform.Output(g, terraformOptions, "master_ips"), ",")
	c.ServerIPs = serverIPs

	rawAgentIPs := terraform.Output(g, terraformOptions, "worker_ips")
	if rawAgentIPs != "" {
		agentIPs = strings.Split(rawAgentIPs, ",")
	}
	c.AgentIPs = agentIPs

	rawAgentWinIPs := terraform.Output(g, terraformOptions, "windows_worker_ips")
	if rawAgentWinIPs != "" {
		winAgentIPs = strings.Split(rawAgentWinIPs, ",")
	}
	c.WinAgentIPs = winAgentIPs

	return c, nil
}
```